docs/source/tutorial-pyfunc.rst

```diff
@@ -0,0 +1,368 @@
+
+.. code-block:: python
+
+    class IrisClassifier(mlflow.pyfunc.PythonModel):
+        def load_context(self, context : mlflow.pyfunc.PythonModelContext) -> None:
+            self.classifier = jl.load(context.artifacts["classifier"])
+
+        def predict(self, context : mlflow.pyfunc.PythonModelContext, model_input : pd.DataFrame) -> np.ndarray[int]:
+            preds = self.classifier.predict(model_input)
+            return iris_classes(preds)
+
+
+    def store_pyfunc_model(classifier : LogisticRegression, signature : ModelSignature) -> str:
+        # MLflow Tracking URI *must* be http or https for log_model
+        assert(mlflow.get_tracking_uri().startswith("http"))
+
+        classifier_filename = "iris_classifier.joblib"
+        jl.dump(classifier, classifier_filename)
+        artifacts = {
+            "classifier": classifier_filename
+        }
+
+        model_info = mlflow.pyfunc.log_model(
+            python_model=IrisClassifier(),
+            artifacts=artifacts,
+            artifact_path="artifacts",
+            signature=signature,
+            code_path= ["custom_code.py"]
+        )
+        print(f"Stored in mlruns/{model_info.run_id}")
+        return model_info.run_id
+
+    if __name__ == "__main__":
+        with mlflow.start_run() as run:
+            classifier, signature = initialize_and_train()
+            run_id = store_pyfunc_model(classifier, signature)
+
+
+Starting at the entry point, you can see that the code does not change the creation of ``classifier`` and ``signature``. These values are passed to ``store_pyfunc_model``. 
+
+The first thing that ``store_pyfunc_model`` does is confirm that the MLflow Tracking Server is running on a URI that starts with ``http`` or ``https``, as discussed in :ref:`setup`.
+
+The code then uses the ``joblib`` library to serialize the model to disk. Using serialization/deserialization to save and load the model and related files can make for clear and efficient code, but it is not required. You can store any type of file in the ``artifacts`` directory: JSON files, binary weight files, vector database files, and so forth. 
+
+The ``artifacts`` dictionary is then created. Its ``classifier`` key is set to the name of the file containing the serialized model: ``"iris_classifier.joblib"``.
+
+An instance of the ``IrisClassifier`` class is then instantiated. This class derives from :py:class:`mlflow.pyfunc.PythonModel`. As discussed at the beginning of this section, this class implements the ``load_context`` and ``predict`` methods. The ``IrisClassifier`` implementation of those methods will be discussed in the next section. 
+
+After the ``IrisClassifier`` instance is created, the ``mlflow.pyfunc.log_model`` method is called. This method takes the following arguments:
+
+- ``python_model``: The instance of the ``IrisClassifier`` class we just created.
+- ``artifacts``: The dictionary containing the ``"classifier"`` key and the ``"iris_classifier.joblib"`` value.
```

Comment:
Can we add a note or a paragraph somewhere near here explaining how important this is and how it is particularly important for models that are massive in size? Referring to an artifact location on the artifact store is critical for situations where the model is truly massive and shouldn't be included within the directly logged model itself (which is what will happen iff the `artifacts` dict is not specified? 
Perhaps a note callout to draw attention to this behavior is warranted.